examples/bond/scripts/run/common/common.js

```diff
@@ -68,6 +68,209 @@ async function optInTx (deployer, managerAcc, lsig, assetIndex) {
+      toAccountAddr: issuerLsig.address(),
+      amount: 1e6,
+      assetID: bondID,
+      payFlags: { }
+    },
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      appID: appID,
+      payFlags: {},
+      appArgs: ['str:issue']
+    }
+  ];
+}
+
+/**
+ * Returns create dex transaction
+ * @param managerAccount App manager account
+ * @param appID application index
+ * @param issuerLsig issuer logic signature
+ * @param dexLsig dex logic signature
+ * @param creatorAccount bond token creator account
+ * @param assetAmount balance of oldBond tokens in issuer address
+ * @param newBondID new bond token index
+ * @param oldBondID old bond token index
+ * @param total total number of bonds issued to buyers
+ * @param couponValue value of coupon
+ */
+function createDexTx (
+  managerAccount, appID, issuerLsig,
+  dexLsig, creatorAccount, assetAmount,
+  newBondID, oldBondID, total, couponValue
+) {
+  return [
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: managerAccount,
+      appID: appID,
+      payFlags: {},
+      appArgs: ['str:create_dex'],
+      accounts: [issuerLsig.address(), dexLsig.address()]
+    },
+    // New bond token transfer to issuer's address
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      toAccountAddr: issuerLsig.address(),
+      amount: assetAmount,
+      assetID: newBondID,
+      payFlags: { totalFee: 1000 }
+    },
+    // burn tokens
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: issuerLsig.address(),
+      lsig: issuerLsig,
+      toAccountAddr: creatorAccount.addr,
+      amount: assetAmount,
+      assetID: oldBondID,
+      payFlags: { totalFee: 1000 }
+    },
+    // Transfer app.total amount of new Bonds to dex lsig
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      toAccountAddr: dexLsig.address(),
+      amount: total,
+      assetID: newBondID,
+      payFlags: { totalFee: 1000 }
+    },
+    // Algo transfer to dex address
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      toAccountAddr: dexLsig.address(),
+      amountMicroAlgos: Number(total) * Number(couponValue),
+      payFlags: { totalFee: 1000 }
+    }
+  ];
+}
+
+/**
+ * Returns redeem transaction
+ * @param buyerAccount buyer account
+ * @param dexLsig dex logic signature
+ * @param amount amount of bonds to be redeemed
+ * @param oldBondID old bond token index
+ * @param newBondID new bond token index
+ * @param couponValue value of coupon
+ * @param appID application index
+ */
+function redeemTx (
```

Comment:
let's rename `redeemTx` to `redeemCouponTx`.

Reply:
ok